cmd/zpool/zpool_main.c

```diff
@@ -676,6 +676,20 @@ zpool_do_add(int argc, char **argv)
 		return (1);
 	}
 
+	/* unless manually specified use "ashift" pool property for new vdevs */
+	if (!nvlist_exists(props, ZPOOL_CONFIG_ASHIFT)) {
```

Comment:
> The pool property cannot be changed, so this would be rather annoying. I don't see it serving much purpose as it is now

Well, without this patch it cannot be changed and does not affect new VDEVs, so i agree it's not doing much. With this patch it is at least used when adding new devices, which is what users seem to expect from this property.

> so we should be able to change it to stop being a onetime thing. Then this would make more sense.
> The situation that I envision is one where the pool autodetected correctly and then you want to set a default ashift for all future vdevs. That will not be allowed.

Changing this property from `PROP_ONETIME` to `PROP_DEFAULT` seems trivial, but i don't see how we would benefit from being able to change it after the pool is created: what's the use case here?